tests/ci/fast_test_check.py

```diff
@@ -151,12 +153,17 @@ def main():
         os.makedirs(logs_path)
 
     run_log_path = os.path.join(logs_path, "run.log")
-    with TeePopen(run_cmd, run_log_path, timeout=90 * 60) as process:
-        retcode = process.wait()
-        if retcode == 0:
-            logging.info("Run successfully")
-        else:
-            logging.info("Run failed")
+    expired_timeout = None
+    with TeePopen(run_cmd, run_log_path, timeout=65) as process:
+        try:
+            retcode = process.wait()
+            if retcode == 0:
```

Comment:
Just check `process.timeout_exceeded` here

Reply:
I'm not sure if `timeout_exceeded` is necessary at all. When we call `wait()` subprocess.TimeoutExpried is thrown in case of timeout. If we want to handle timeout via the flag `timeout_exceeded` then we should handle the exception in wait(). That's what I'd expect

Reply:
No it's not. See the code of TeePopen. We don't use subprocess timeout, but code that much more wreckless and kills everything without throwing any exceptions

```
In [44]: with TeePopen('sleep 20', log_file='/dev/null', timeout=10) as process:
    ...:     process.wait()
    ...:     pr = process
    ...:
WARNING:root:Killing process 1304088, timeout 10 exceeded
Command `sleep 20` has failed, timeout 10s is exceeded
In [45]: pr
Out[45]: <tee_popen.TeePopen at 0x7f3ef540bf10>

In [46]: pr.timeout_exceeded
Out[46]: True
```

Reply:
[_check_timeout](https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/tee_popen.py#L34-L46) is launched in [thread](https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/tee_popen.py#L59-L62) if the timeout is set. And no exceptions are thrown.